From: "Barak A. Pearlmutter" <barak+git@pearlmutter.net>
Date: Mon, 3 Oct 2022 19:38:46 +0100
Subject: i686 kernel version

On arch i686, there is code to fail for kernel version <2.4, but it's
written as a case statement that ends at version 5.x. The test is
altered to allow arbitrarily kernel versions >=2.4.
---
 config/_arch-n-opsys | 12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)
 mode change 100644 => 100755 config/_arch-n-opsys

diff --git a/config/_arch-n-opsys b/config/_arch-n-opsys
old mode 100644
new mode 100755
index c49b3a5..dfcd8f4
--- a/config/_arch-n-opsys
+++ b/config/_arch-n-opsys
@@ -102,15 +102,9 @@ case `uname -s` in
       *86)
 	ARCH=x86
       # we no longer support Linux before the 2.4 kernel.
-	case `uname -r` in
-	  2.4.*) ;;
-	  2.5.*) ;;
-	  2.6.*) ;;
-	  3.*) ;;
-	  4.*) ;;
-	  5.*) ;;
-	  *) exit 1 ;;
-	esac
+	if [ $(uname -r) '<' 2.4 ]; then
+	    exit 1
+	fi
 	;;
       x86_64)
 	ARCH=$(pick_arch x86 amd64)
